tests/test_erased.rs

```diff
@@ -10,10 +10,9 @@ struct Hello {
 }
 
 fn hello() -> Box<dyn Annotate> {
-    let hello = Box::new(Hello {
+    Box::new(Hello {
         message: "Hello World!".into(),
-    });
-    hello
+    }) as _
```

Comment:
What's ` as _` for here?

Reply:
Good question. `as _` is normally used to introduce a coercion site. Rust provides sane default in cases where coercion site is hardly needed, but here's some cases where it is necessary:
```rust
fn no_coercion_site(x: &Arc<u32>) -> Arc<dyn Any> {
    // This won't work because type inference inhibits the implicit coercion site
    Arc::clone(x)
}

fn explicit_coercion_site(x: &Arc<u32>) -> Arc<dyn Any> {
    // Explicit coercion site provided
    Arc::clone(x) as _
}

fn implicit_coercion_site(x: &Arc<u32>) -> Arc<dyn Any> {
    // Function return type has an implicit coercion site. Method syntax makes
    // compiler infer that this must be `<Arc<u32> as Clone>::clone`
    x.clone()
}
```

I see why clippy is being conservative and always generate a `_`. In this particular case though, you're correct that we don't need `as _`.